### PR TITLE
feat(sdk): support HTTP/S PROXY for SDK client 

### PR DESCRIPTION
**Pull request by rmgogogo**
**Description of your changes:**

support HTTP/S PROXY for SDK client

#4214 

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

File: sdk/python/kfp/_client.py [hunk 1]
```diff
@@ -98,7 +98,8 @@ class Client(object):
   LOCAL_KFP_CONTEXT = os.path.expanduser('~/.config/kfp/context.json')
 
   # TODO: Wrap the configurations for different authentication methods.
-  def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client_id=None, other_client_secret=None, existing_token=None, cookies=None):
+  def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client_id=None, other_client_secret=None, existing_token=None):
+  def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client_id=None, other_client_secret=None, existing_token=None, cookies=None, proxy=None, ssl_ca_cert=None):
```
Comment (numerology) [inline sdk/python/kfp/_client.py:102]: drive-by comment:

It seems like the interface of kfp.Client() is bloated because of various types of connections/deployments we have. That makes me think automatically inferring the right behavior of the authentication will be more and more difficult. 

Should we be more explicit and ask the users to specify the type of connection here? That will make the behavior more predictable.

Comment (rmgogogo) [reply sdk/python/kfp/_client.py:101]: the problem is that we don't know those cases before CE reports it, e.x. this proxy case.
The usage will be

client=kfp.Client(host=xxx, proxy=yyy,ssl_ca_cert=zzz) which seems OK.

For your concern, another solution (maybe better) is that we allow directly pass in configuration object. It doesn't solve the authentication (this also also is not), but it helps for advanced network connections.

**Comment by rmgogogo** [comment]
this PR is ready to go.
one customer verified it in proxy env.

File: sdk/python/kfp/_client.py (hunk 1 above)
Comment (numerology) [reply sdk/python/kfp/_client.py:101]: Sounds good. I am in favor of being more explicit in this case. Indeed it will be very hard to cover all possible cases on our end exhaustively.
